scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
Regarding backwards incompatibilities: If the format option is changed, the actual output file is different. But regardless of the format option the file can still be red. 
I also think that a format option like `fmt='%.4f'` is the best way.
Maybe, @matthew-brett or someone else can merge my PR https://github.com/scipy/scipy/pull/4875 and this bug fix. Afterwards,  I can restart from there with adding the format option and make a new PR.